static/src/javascripts/projects/common/modules/crosswords/helpers.js

```diff
@@ -0,0 +1,364 @@
+// @flow
+import { constants } from 'common/modules/crosswords/constants';
+import findIndex from 'lodash/arrays/findIndex';
+import flatten from 'lodash/arrays/flatten';
+import range from 'lodash/arrays/range';
+import uniq from 'lodash/arrays/uniq';
+
+const isAcross = (clue: Clue): boolean => clue.direction === 'across';
+
+const getLastCellInClue = (clue: Clue): Position => {
+    const ax = {
+        true: 'x',
+        false: 'y',
+    };
+    const axis = ax[String(isAcross(clue))];
+    const otherAxis = ax[String(!isAcross(clue))];
+    const cell: Position = ({
+        [axis]: clue.position[axis] + (clue.length - 1),
+        [otherAxis]: clue.position[otherAxis],
+    }: any);
+
+    return cell;
+};
+
+const isFirstCellInClue = (cell: Position, clue: Clue): boolean => {
+    const axis = isAcross(clue) ? 'x' : 'y';
+
+    return cell[axis] === clue.position[axis];
+};
+
+const isLastCellInClue = (cell: Position, clue: Clue): boolean => {
+    const axis = isAcross(clue) ? 'x' : 'y';
+
+    return cell[axis] === clue.position[axis] + (clue.length - 1);
+};
+
+const getNextClueInGroup = (entries: Array<Clue>, clue: Clue): ?Clue => {
+    const newClueId =
+        clue.group[findIndex(clue.group, id => id === clue.id) + 1];
+
+    return entries.find(entry => entry.id === newClueId);
+};
+
+const getPreviousClueInGroup = (entries: Array<Clue>, clue: Clue): ?Clue => {
+    const newClueId =
+        clue.group[findIndex(clue.group, id => id === clue.id) - 1];
+
+    return entries.find(entry => entry.id === newClueId);
+};
+
+const getGroupEntriesForClue = (
+    entries: Array<Clue>,
+    group: Array<string>
+): Array<Clue> =>
+    group.reduce((acc, clueId) => {
+        const entry = entries.find(e => e.id === clueId);
```

Comment:
I have changed this from a `map` to a `reduce` as the above `find` call could theoretically return `undefined`